pkg/aws/awscontrolplane/validate_awscontrolplane_test.go

```diff
@@ -149,6 +151,54 @@ func TestAZValid(t *testing.T) {
 			fakeK8sClient := unittest.FakeK8sClient()
 			validate := &Validator{
 				validAvailabilityZones: tc.validAZs,
+				validInstanceTypes:     unittest.DefaultInstanceTypes(),
+				k8sClient:              fakeK8sClient,
+				logger:                 microloggertest.New(),
+			}
+
+			admissionRequest, err := unittest.DefaultAdmissionRequestAWSControlPlane()
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			allowed, _ := validate.Validate(&admissionRequest)
+			if allowed != tc.allowed {
+				t.Fatalf("expected %v to not to differ from %v", allowed, tc.allowed)
+			}
+		})
+	}
+}
+func TestInstanceTypeValid(t *testing.T) {
```

Comment:
As a general note about unit tests. I noticed the tests work with the same input object in all tests. I feel like this makes it harder to comprehend the unit test itself. It also does not represent real world usage. In real world the admission controller config is rather static and the CRs to be validated change all the time. The test is the other way around. Not saying one approach is better than the other. Just sharing an observation.

Reply:
Yeah, I agree.
We have made it much easier to manipulate the config than the admission request so the quickest test we can write at the moment is one like this. It's something I'd like to improve by adding to the `unittest` package.